crates/ruff_cli/src/args.rs

```diff
@@ -365,6 +365,14 @@ pub struct FormatCommand {
     /// Path to the `pyproject.toml` or `ruff.toml` file to use for configuration.
     #[arg(long, conflicts_with = "isolated")]
     pub config: Option<PathBuf>,
+
+    /// Disable cache reads.
+    #[arg(short, long, help_heading = "Miscellaneous")]
+    pub no_cache: bool,
+    /// Path to the cache directory.
+    #[arg(long, env = "RUFF_CACHE_DIR", help_heading = "Miscellaneous")]
+    pub cache_dir: Option<PathBuf>,
```

Comment:
I think we would need to have different environment variables (`RUFF_CHECK_CACHE_DIR`, `RUFF_FORMAT_CACHE_DIR`) since one could potentially provide different path to the `check` / `format` subcommand.

Or, could make this a global flag?

Reply:
I would wait until we have a specific request to store the caches in different locations. I also don't know how to support different cache locations when `check` lints and formats. 

The alternative would be to have two different caches (the implementation I had before this morning) where `format` and `lint` store their results in different files. Different jobs could then cache the specific sub-directories only. However, the overall design of the `Cache` API felt more fragile and would probably require a more drastic refactor.

Reply:
Not necessary IMO.